### PR TITLE
VisualStudio™ build

### PR DESCRIPTION
**Pull request by isage**
Welp, here it is. I've also fixed couple of undefined-behavior bugs and one bug in 2.1 render.
Due to drop of our own make_unique implementation build now requires gcc 4.9/clang 3.5 or later.
Should be available in any recent distro, though.

I'll post build tutorial(s) after merge.

### CODE REVIEW

File: src/Format/Ini/File.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ Section& File::section(const std::string &name)
     if (it == _sectionIdxMap.end())
     {
         auto idx = _sections.size();
-        _sections.push_back(std::move(Section(name)));
+        _sections.push_back(Section(name));
```
Comment (dreamer-dead) [inline src/Format/Ini/File.cpp]: Maybe just 
_sections.emplace_back(name);
?

File: src/Format/Ini/Value.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ int Value::toInt() const
     {
         return std::stoi(_value);
     }
-    catch (std::invalid_argument)
```
Comment (dreamer-dead) [inline src/Format/Ini/Value.cpp]: I'm interested, why exceptions often handled by non-const reference?
Will it be changed? =)

File: src/Format/Ini/File.cpp (hunk 1 above)
Comment (isage) [inline src/Format/Ini/File.cpp]: Sure

File: src/Format/Ini/Value.cpp (hunk 2 above)
Comment (isage) [inline src/Format/Ini/Value.cpp]: Not that it's really important, but yep, sure.

File: src/PathFinding/HexagonGrid.cpp [hunk 3]
```diff
@@ -247,7 +245,7 @@ Hexagon* HexagonGrid::hexInDirection(Hexagon* from, unsigned short rotation, uns
 std::vector<Hexagon*> HexagonGrid::ring(Hexagon* from, unsigned int radius)
 {
     std::vector<Hexagon*> result;
-    Hexagon* current = nullptr;
+    Hexagon* current;
```
Comment (dreamer-dead) [inline src/PathFinding/HexagonGrid.cpp]: Why this change?
We can just move down this local variable to line 255.

Comment (isage) [inline src/PathFinding/HexagonGrid.cpp]: Two consecutive assignments. But yeah, it should be moved down. Done.

**Comment by isage** [comment]
Btw, we also tested this branch on openbsd/freebsd, works there too.

File: src/ResourceManager.cpp [hunk 4]
```diff
@@ -403,8 +402,7 @@ Graphics::Shader* ResourceManager::shader(const string& filename)
         return _shaders.at(filename).get();
     }
 
-    Graphics::Shader* shader = nullptr;
-    shader = new Graphics::Shader(filename);
+    Graphics::Shader* shader = new Graphics::Shader(filename);
 
     _shaders.insert(make_pair(filename, unique_ptr<Graphics::Shader>(shader)));
```
Comment (dreamer-dead) [inline src/ResourceManager.cpp]: It seems that we can use emplace() method like
_shaders.emplace(filename, make_unique<Graphics::Shader>(filename)));

File: src/UI/MvePlayer.cpp [hunk 5]
```diff
@@ -837,8 +819,10 @@ void MvePlayer::_processChunk()
         return;
     }
 
-    for (auto opcode : *_chunk->opcodes())
+    auto opcodes = _chunk->opcodes();
+    for (unsigned int i=0; i < opcodes->size();i++)
```
Comment (dreamer-dead) [inline src/UI/MvePlayer.cpp]: Can we use range-based for loop here? Like 

`for (const auto& opcode : opcodes) ...`

File: src/Format/Ini/Value.cpp (hunk 2 above)
Comment (dreamer-dead) [inline src/Format/Ini/Value.cpp]: That is a common practice, I see a lot of code with that approach from different developers.
But this is quite strange for me =)

File: src/ResourceManager.cpp (hunk 4 above)
Comment (isage) [inline src/ResourceManager.cpp]: Yeah, but we still need to return shader.

Comment (dreamer-dead) [inline src/ResourceManager.cpp]: This makes sense, thanks for clarification.

File: src/UI/MvePlayer.cpp (hunk 5 above)
Comment (isage) [inline src/UI/MvePlayer.cpp]: _chunk->opcodes() returns pointer, so
````
auto opcodes = *_chunk->opcodes();
for (const auto& opcode : opcodes)
````
Strangely, first time i tried, VS croaked with error.

Comment (dreamer-dead) [inline src/UI/MvePlayer.cpp]: Be careful, because

`auto opcodes = *_chunk->opcodes();`
will make a copy of the vector.
`const auto& opcodes = *_chunk->opcodes();`
is a preferred way.

Comment (isage) [inline src/UI/MvePlayer.cpp]: Oh, right. Fixed.
